src/main/java/com/google/devtools/build/lib/analysis/test/TestConfiguration.java

```diff
@@ -54,13 +57,21 @@ public class TestConfiguration extends Fragment {
           // changes in --trim_test_configuration itself or related flags always prompt invalidation
           return true;
         }
+        // LINT.IfChange
         Class<? extends FragmentOptions> affectedOptionsClass =
             changedOption.getDeclaringClass(FragmentOptions.class);
         if (!affectedOptionsClass.equals(TestOptions.class)
             && !affectedOptionsClass.equals(CoverageOptions.class)) {
-          // options outside of TestOptions always prompt invalidation
+          // options outside of TestOptions always prompt invalidation, except for --run_under.
+          if (affectedOptionsClass.equals(CoreOptions.class)
+              && changedOption.getOptionName().equals("run_under")) {
```

Comment:
Is it worth moving `--run_under` to a new `RunUnderOptions` class? Would that simplify things or make them more complicated?

Reply:
I think that this doesn't simplify the situation as it would just introduce another one-of class that is involved in test trimming (`CoverageOptions` was already missed here). 

If we want to simplify this check, we could instead replace the `OptionDiffPredicate` concept with something less indirect. For example, we could just apply `TestTrimmingLogic.trim` to the old and new options and take the `OptionDiff` of that. That's almost straightforward except that we also need to take changes to `--experimental_retain_test_configuration_across_testonly` into account. I didn't want to do this in the current change.

Reply:
Sounds reasonable, thanks for clarifying that.